File: src/main/java/com/crossborder/shop/entity/Order.java
package com.crossborder.shop.entity;

import com.crossborder.shop.common.BaseEntity;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 订单实体
 *
 * @author CrossBorder Shop
 * @since 2026-02-04
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class Order extends BaseEntity {

    /**
     * 订单号
     */
    private String orderNumber;

    /**
     * 买家ID
     */
    private Long buyerId;

    /**
     * 卖家ID
     */
    private Long sellerId;

    /**
     * 订单状态：0-待支付，1-待发货，2-已发货，3-已完成，4-已取消，5-已退款
     */
    private Integer orderStatus;

    /**
     * 订单总金额
     */
    private BigDecimal totalAmount;

    /**
     * 商品总金额
     */
    private BigDecimal productAmount;

    /**
     * 运费
     */
    private BigDecimal freightAmount;

    /**
     * 优惠金额
     */
    private BigDecimal discountAmount;

    /**
     * 原币种
     */
    private String currency;

    /**
     * 汇率
     */
    private BigDecimal exchangeRate;

    /**
     * 换算后金额（目标币种）
     */
    private BigDecimal convertedAmount;

    /**
     * 目标币种
     */
    private String targetCurrency;

    /**
     * 支付状态：0-未支付，1-已支付，2-已退款
     */
    private Integer paymentStatus;

    /**
     * 支付方式
     */
    private String paymentMethod;

    /**
     * 支付时间
     */
    private LocalDateTime paymentTime;

    /**
     * 支付流水号
     */
    private String paymentTransactionId;

    /**
     * 发货时间
     */
    private LocalDateTime shipTime;

    /**
     * 完成时间
     */
    private LocalDateTime completeTime;

    /**
     * 取消时间
     */
    private LocalDateTime cancelTime;

    /**
     * 取消原因
     */
    private String cancelReason;

    /**
     * 备注
     */
    private String remark;

    /**
     * 买家留言
     */
    private String buyerMessage;

    /**
     * 版本号
     */
    private Integer version;
}

File: .shop/src/models/order.ts
// Order related types
export interface Order {
  id: number;
  orderNumber: string; // 后端字段名
  buyerId: number; // 买家ID
  sellerId?: number; // 卖家ID
  orderStatus: OrderStatus; // 订单状态
  paymentStatus?: PaymentStatus; // 支付状态
  totalAmount: number; // 订单总金额
  productAmount?: number; // 商品总金额
  freightAmount?: number; // 运费
  discountAmount?: number; // 优惠金额
  currency?: string; // 原币种
  exchangeRate?: number; // 汇率
  convertedAmount?: number; // 换算后金额
  targetCurrency?: string; // 目标币种
  paymentMethod?: string; // 支付方式
  paymentTime?: string; // 支付时间
  paymentTransactionId?: string; // 支付流水号
  logisticsCompany?: string; // 物流公司
  trackingNo?: string; // 物流单号
  sellerAddress?: string; // 卖家退货地址
  shipTime?: string; // 发货时间
  completeTime?: string; // 完成时间
  cancelTime?: string; // 取消时间
  cancelReason?: string; // 取消原因
  remark?: string; // 备注
  buyerMessage?: string; // 买家留言
  createTime?: string; // 创建时间
  items?: OrderItem[]; // 订单明细
  address?: ShippingAddress; // 收货地址
}

export enum OrderStatus {
  PENDING = 0,
  PAID = 1,
  SHIPPED = 2,
  COMPLETED = 3,
  CANCELED = 4,
  REFUNDING = 5,
  REFUNDED = 6,
}

export enum PaymentStatus {
  UNPAID = 0,
  PAID = 1,
  REFUNDED = 2,
}

export enum ShippingStatus {
  PENDING = 0,
  SHIPPED = 1,
  DELIVERED = 2,
}

export interface OrderItem {
  id: number;
  orderId: number;
  productId: number;
  productName: string;
  productImage?: string;
  price: number;
  quantity: number;
  totalAmount: number;
}

export interface OrderListParams {
  page?: number;
  pageSize?: number;
  orderNo?: string;
  userId?: number;
  status?: OrderStatus;
  paymentStatus?: PaymentStatus;
  shippingStatus?: ShippingStatus;
  startDate?: string;
  endDate?: string;
  minAmount?: number;
  maxAmount?: number;
}

export interface OrderListResponse {
  list: Order[];
  total: number;
  page: number;
  pageSize: number;
}

export interface ShippingAddress {
  id: number;
  userId?: number;
  receiverName: string;
  receiverPhone: string;
  province: string;
  city: string;
  district: string;
  detailAddress: string;
  isDefault?: boolean;
  label?: string;
}

export interface OrderStatusUpdateDTO {
  status: OrderStatus;
  remark?: string;
}

export interface OrderShipDTO {
  trackingNo?: string;
  shippingCompany?: string;
  shippingCompanyCode?: string;
  remark?: string;
}

export interface RefundDTO {
  reason: string;
  amount?: number;
  images?: string[];
}
